executor/runtime/docker/docker.go

```diff
@@ -1163,55 +1165,49 @@ func (r *DockerRuntime) Prepare(ctx context.Context, pod *v1.Pod) error { // nol
 		bindMounts = append(bindMounts, getKernelBindMounts()...)
 	}
 
-	dockerCfg, hostCfg, err = r.mainContainerDockerConfig(r.c, bindMounts, size, volumeContainers)
+	dockerCfg, hostCfg, err := r.mainContainerDockerConfig(r.c, bindMounts, imageSize, volumeContainers)
 	if err != nil {
-		goto error
+		return err
 	}
 
 	logger.G(ctx).WithFields(map[string]interface{}{
 		"dockerCfg": logger.ShouldJSON(ctx, *dockerCfg),
 		"hostCfg":   logger.ShouldJSON(ctx, *hostCfg),
 	}).Info("Creating container in docker")
 
-	containerCreateBody, err = r.client.ContainerCreate(ctx, dockerCfg, hostCfg, nil, r.c.TaskID())
+	containerCreateBody, err := r.client.ContainerCreate(ctx, dockerCfg, hostCfg, nil, r.c.TaskID())
 	r.c.SetID(containerCreateBody.ID)
 
 	r.metrics.Timer("titus.executor.dockerCreateTime", time.Since(dockerCreateStartTime), r.c.ImageTagForMetrics())
 	if docker.IsErrNotFound(err) {
 		return &runtimeTypes.RegistryImageNotFoundError{Reason: err}
 	}
 	if err != nil {
-		goto error
+		return err
```

Comment:
can we use tracehelpers.setstatus on these error messages?

Reply:
I don't know much about what that does, but we already have that on the `defer`, isn't that enough?

Reply:
For example, any `return err` that is non nil in this function should call

    tracehelpers.SetStatus(err, span)